test/python/compiler/test_disassembler.py

```diff
@@ -221,6 +235,151 @@ def test_multiple_conditionals_multiple_registers(self):
         self.assertEqual(circuits[0], qc)
         self.assertEqual({}, header)
 
+    def assertCircuitCalibrationsEqual(self, circuits, output_circuits):
+        """Verify circuit calibrations are equivalent pre-assembly and post-disassembly"""
+        self.assertEqual(
+            all(
+                len(qc.calibrations) == len(out_qc.calibrations)
+                for qc, out_qc in zip(circuits, output_circuits)
+            ),
+            True,
+        )
+        self.assertEqual(
+            all(
+                circuit.calibrations.keys() == out_qc.calibrations.keys()
+                for circuit, out_qc in zip(circuits, output_circuits)
+            ),
+            True,
+        )
+        self.assertEqual(
+            {tuple(qc_cal.keys()) for qc in circuits for qc_cal in qc.calibrations.values()},
+            {
+                tuple(out_qc_cal.keys())
+                for out_qc in output_circuits
+                for out_qc_cal in out_qc.calibrations.values()
+            },
+        )
+        self.assertEqual(
+            all(
+                _parametric_to_waveforms(qc_sched) == _parametric_to_waveforms(out_qc_sched)
```

Comment:
Why convert to waveforms? Were you having an issue when you didn't do this?

Reply:
if the source calibration is a parametric pulse but the backend can't handle that type of pulse, then the output calibration will be a waveform. to compare that source and output are equal we need to convert one form to the other , and since we cant convert waveform to parametric easily, we convert both to waveforms to check equality if that makes sense.

![Screen Shot 2021-05-19 at 7 49 32 PM](https://user-images.githubusercontent.com/25019973/118898777-56e30800-b8db-11eb-8a87-f39e9fed932b.png)

Reply:
That does make sense. Could we add one test (or an additional line in one of the existing tests) that if the backend does support parametric pulses, the disassembled pulses are also parametric?

Reply:
yup of course, added a direct equality check to `test_parametric_pulse_circuit_calibrations`